packages/intl-getcanonicallocales/BUILD

```diff
@@ -0,0 +1,167 @@
+load("@build_bazel_rules_nodejs//:index.bzl", "generated_file_test", "npm_package_bin", "pkg_npm")
+load("@npm//@bazel/rollup:index.bzl", "rollup_bundle")
+load("@npm//@bazel/typescript:index.bzl", "ts_project")
+load("@npm//karma:index.bzl", "karma_test")
+load("//tools:jest.bzl", "jest_test")
+
+PACKAGE_NAME = "intl-getcanonicallocales"
+
+pkg_npm(
+    name = PACKAGE_NAME,
+    srcs = ["package.json"],
+    deps = [":dist"],
+)
+
+SRCS = glob(["src/*.ts"]) + [
+    "index.ts",
+    "polyfill.ts",
+]
+
+TESTS = glob([
+    "tests/*.test.ts",
+])
+
+SRC_DEPS = [
+    "@npm//cldr-core",
+    "@npm//@types/node",
+]
+
+TEST_DEPS = SRC_DEPS + [
+    "@npm//@jest/transform",
+    "@npm//ts-jest",
+    "@npm//@types/jest",
+]
+
+ts_project(
+    name = "dist",
+    package_name = "@formatjs/%s" % PACKAGE_NAME,
+    srcs = SRCS,
+    declaration = True,
+    declaration_map = True,
+    extends = ["//:tsconfig.json"],
+    source_map = True,
+    tsconfig = "tsconfig.json",
+    visibility = ["//visibility:public"],
+    deps = SRC_DEPS,
+)
+
+ts_project(
+    name = "lib",
+    package_name = "@formatjs/%s" % PACKAGE_NAME,
+    srcs = SRCS,
+    declaration = True,
+    declaration_map = True,
+    extends = ["//:tsconfig.json"],
+    outdir = "lib",
+    source_map = True,
+    tsconfig = "tsconfig.json",
+    visibility = ["//visibility:public"],
+    deps = SRC_DEPS,
+)
+
+jest_test(
+    name = "unit",
+    srcs = [
+        "//:tsconfig.json",
+    ] + SRCS + TESTS,
+    jest_config = ":jest.config.js",
+    tags = [
+        # Need to set the pwd to avoid jest needing a runfiles helper
+        # Windows users with permissions can use --enable_runfiles
+        # to make this test work
+        "no-bazelci-windows",
+        # TODO: why does this fail almost all the time, but pass on local Mac?
+        "no-bazelci-mac",
+    ],
+    deps = TEST_DEPS,
+)
+
+# aliases
+npm_package_bin(
+    name = "gen_aliases",
+    outs = ["tmp/aliases.ts"],
+    args = [
+        "$(execpath scripts/aliases.ts)",
+        "--out",
+        "$@",
+    ],
+    data = [
+        "scripts/aliases.ts",
+        "//:tsconfig.json",
+        "@npm//@types/fs-extra",
+        "@npm//@types/minimist",
+        "@npm//@types/node",
+        "@npm//cldr-core",
+        "@npm//fs-extra",
+        "@npm//minimist",
```

Comment:
Can these be automatically inferred from import statements or just package.json? What are supposed to be put inside here?